build/configuration-properties-collector/src/main/java/org/hibernate/search/configuration/properties/collector/ConfigurationPropertyCollector.java

```diff
@@ -0,0 +1,211 @@
+				constant,
+				classPrefix,
+				annotation.flatMap( a -> a.multiAttribute( "prefix", String.class ) )
+		);
+		if ( !key.matches( ignoreKeys ) ) {
+			// Try to find a default value. Assumption is that the settings class has an inner class called "Defaults" and
+			// the key for the default value is exactly the same as the config constant name:
+			Object value = findDefault( constant );
+
+			properties.put(
+					constant.getEnclosingElement().toString() + "#" + constant.getSimpleName().toString(),
+					new ConfigurationProperty()
+							.javadoc( extractJavadoc( constant ) )
+							.key( key )
+							.sourceClass( constant.getEnclosingElement().toString() )
+							.type( extractType( constant ) )
+							.defaultValue( value )
+			);
+		}
+	}
+
+
+	private ConfigurationProperty.Key extractKey(VariableElement constant, Optional<List<String>> classPrefix, Optional<List<String>> constantPrefix) {
+		List<String> prefix;
+		if ( constantPrefix.isPresent() ) {
+			prefix = constantPrefix.get();
+		}
+		else if ( classPrefix.isPresent() ) {
+			prefix = classPrefix.get();
+		}
+		else {
+			prefix = configPrefix;
+		}
+
+		return new ConfigurationProperty.Key(
+				prefix,
+				Objects.toString( constant.getConstantValue(), "NOT_FOUND#" + constant.getSimpleName() )
+		);
+	}
+
+	private HibernateSearchConfiguration.Type extractType(VariableElement constant) {
+		return constant.getEnclosingElement().getEnclosingElement()
+				.toString().contains( ".spi" ) ?
+				HibernateSearchConfiguration.Type.SPI :
+				HibernateSearchConfiguration.Type.API;
+	}
+
+	private String extractJavadoc(VariableElement constant) {
+		try {
+			Element enclosingClass = constant.getEnclosingElement();
+			Path docs = javadocsLocation.resolve(
+					enclosingClass.toString().replace( ".", File.separator ) + ".html"
+			);
+
+			// calling getEnclosingElement() on class should return a package:
+			String packagePath = enclosingClass.getEnclosingElement().toString().replace( ".", File.separator );
+
+			Document javadoc = Jsoup.parse( docs.toFile() );
+
+			org.jsoup.select.Elements blocks = javadoc.select( "#" + constant.getSimpleName() + " .block" );
+			if ( !blocks.isEmpty() ) {
+				org.jsoup.nodes.Element block = blocks.get( 0 );
+				for ( org.jsoup.nodes.Element link : block.getElementsByTag( "a" ) ) {
+					String href = link.attr( "href" );
+					// only update links if they are not external:
+					if ( !link.hasClass( "external-link" ) ) {
+						if ( href.startsWith( "#" ) ) {
+							href = enclosingClass.getSimpleName().toString() + ".html" + href;
+						}
+						href = javadocsBaseLink + packagePath + "/" + href;
+					}
+					else if ( href.contains( "/search/build/parents/" ) && href.contains( "/apidocs" ) ) {
+						// means a link was to a class from other module and javadoc plugin generated some external link
+						// that won't work. So we replace it:
+						href = javadocsBaseLink + href.substring( href.indexOf( "/apidocs" ) + 8 );
```

Comment:
No magic numbers pretty please :x 

```suggestion
						href = javadocsBaseLink + href.substring( href.indexOf( "/apidocs" ) + "/apidocs".length() );
```